src/java.base/share/classes/sun/security/pkcs/PKCS9Attribute.java

```diff
@@ -428,95 +287,18 @@ public PKCS9Attribute(DerValue derVal) throws IOException {
         }
 
         // check single valued have only one value
-        if (SINGLE_VALUED[index] && elems.length > 1)
+        if (info.singleValued() && elems.length > 1)
             throwSingleValuedException();
 
         // check for illegal element tags
-        Byte tag;
+        byte tag;
         for (DerValue elem : elems) {
             tag = elem.tag;
-            if (indexOf(tag, PKCS9_VALUE_TAGS[index], 0) == -1)
+            if (indexOf(tag, info.valueTags(), 0) == -1)
                 throwTagException(tag);
         }
 
-        switch (index) {
-        case 1:     // email address
-        case 2:     // unstructured name
-        case 8:     // unstructured address
-            { // open scope
-                String[] values = new String[elems.length];
-
-                for (int i=0; i < elems.length; i++)
-                    values[i] = elems[i].getAsString();
-                value = values;
-            } // close scope
-            break;
-
-        case 3:     // content type
-            value = elems[0].getOID();
-            break;
-
-        case 4:     // message digest
-            value = elems[0].getOctetString();
-            break;
-
-        case 5:     // signing time
-            byte elemTag = elems[0].getTag();
-            DerInputStream dis = new DerInputStream(elems[0].toByteArray());
-            value = dis.getTime();
-            break;
```

Comment:
The current changes both: 
1. Removes the `elemTag` variable since it is never used and calling `getTag()` does not change `elems`.
2. Simply sets `value = elems[0].getTime()` so as to avoid the conversion to a `DerInputStream` and back.
Neither of these changes caused problems when testing but I wanted to double check before the PR gets merged.